fix/fixer_qemu_disk_size.go

```diff
@@ -0,0 +1,52 @@
+package fix
+
+import (
+	"strconv"
+
+	"github.com/mitchellh/mapstructure"
+)
+
+// FixerQEMUDiskSize updates disk_size from a string to int for QEMU builders
+type FixerQEMUDiskSize struct{}
+
+func (FixerQEMUDiskSize) Fix(input map[string]interface{}) (map[string]interface{}, error) {
+	type template struct {
+		Builders []map[string]interface{}
+	}
+
+	// Decode the input into our structure, if we can
+	var tpl template
+	if err := mapstructure.Decode(input, &tpl); err != nil {
+		return nil, err
+	}
+
+	for _, builder := range tpl.Builders {
+		builderTypeRaw, ok := builder["type"]
+		if !ok {
+			continue
+		}
+
+		builderType, ok := builderTypeRaw.(string)
+		if !ok {
+			continue
+		}
+
+		if builderType != "qemu" {
+			continue
+		}
+
+		diskSize, ok := builder["disk_size"].(int)
```

Comment:
I ran this fixer against a real qemu config where `"disk_size": 5000,` and it didn't work. It turns out that my disk_size was read into the builder struct as a float64, not an int, so this assertion was failing.  

Did you validate against other templates where it was being read as an int?  I tried disk_sizes of 5, 50, 500, and 5000 and they all got read as float64.  If you can reproduce this behavior by running `packer fix` against an actual template, we should change this line to check for the right type:

```suggestion
		diskSize, ok := builder["disk_size"].(float64)
```

Reply:
I was able to repro this. I'm not sure why it gets read in as float64.

I've added tests to explicitly test int and float64 types.